server/src/main/java/org/elasticsearch/http/HttpTransportSettings.java

```diff
@@ -88,6 +88,15 @@
         Setting.byteSizeSetting("http.max_chunk_size", new ByteSizeValue(8, ByteSizeUnit.KB), Property.NodeScope);
     public static final Setting<ByteSizeValue> SETTING_HTTP_MAX_HEADER_SIZE =
         Setting.byteSizeSetting("http.max_header_size", new ByteSizeValue(8, ByteSizeUnit.KB), Property.NodeScope);
+    // Setting the default max count of warning header type to 62, as default max count for all types of headers is 64.
+    // This leaves 2 headers for content-type and content-length.
+    public static final Setting<Integer> SETTING_HTTP_MAX_WARNING_HEADER_COUNT =
+        Setting.intSetting("http.max_warning_header_count", 62, 0, Setting.Property.Dynamic, Property.NodeScope);
+    // Setting the default max count of warning header size to 7Kb, as default max size for all types of headers is 8Kb.
+    // This leaves 1Kb for  headers of content-type and content-length.
```

Comment:
I am not sure what default you're referring to here? As far as I know, there is no default limit enforced by Elasticsearch on header count nor aggregate header size on HTTP responses?

Reply:
@jasontedor Thanks for the comment.  Yes, you are right, there is no default limits enforced by ES. These were the default settings used by Elastic Cloud, and I thought they are more or less universal.  What defaults would you recommend?

Reply:
Philosophically I do not think our defaults should be based on Cloud constraints. It’s a separate product with specific limitations because of the nature of the infrastructure there. Additionally, their constraints can change on a lifecycle independent of our own. Instead, we should choose what makes sense for the wider user base while Cloud can utilize these knobs to tune to their infrastructure. I think about two things here:
 - a bounded default is a behavioral breaking change in a minor release; we should probably not do that unless we consider unbounded a bug
 - I think it is a mistake to silently truncate
 - if a client application blows up because Elasticsearch sends it too large of response headers, this is a problem that is easily searchable for solutions to (where developers of such application would stumble on these settings and tune their Elasticsearch instance in response)

Therefore, I lean towards the default being unbounded. What do you think?